addthis_block/src/Plugin/Block/AddThisBlock.php

```diff
@@ -26,173 +25,109 @@ class AddThisBlock extends BlockBase
-      )
-    );
-    $form['settings']['addthis_settings']['extra_css'] = array(
-      '#title' => t('Extra CSS declaration'),
-      '#type' => 'textfield',
-      '#size' => 40,
-      '#default_value' => $this->configuration['extra_css'],
-      '#description' => t('Specify extra CSS classes to apply to the toolbox'),
+    $form['settings']['addthis_settings']['type_settings'] = array(
+      '#prefix' => '<div id="addthis_type_settings"',
+      '#suffix' => '</div>',
     );
+    if($type == 'addthis_basic_toolbox'){
+      $basicToolbox = AddThis::getInstance()->getBasicToolboxForm($this, $settings['basic_toolbox']);
+      $form['settings']['addthis_settings']['type_settings']['basic_toolbox'] = $basicToolbox;
+    }
+    else if ($type == 'addthis_basic_button'){
+      $basicButton = AddThis::getInstance()->getBasicButtonForm($this, $settings['basic_button']);
+      $form['settings']['addthis_settings']['type_settings']['basic_button'] = $basicButton;
+    }
+
+
     return $form;
   }
-  public function blockSubmit($form, FormStateInterface $form_state) {
-    $this->configuration['share_services'] = $form_state->getValue(['settings', 'addthis_settings', 'share_services']);
-    $this->configuration['buttons_size'] = $form_state->getValue(['settings', 'addthis_settings', 'buttons_size']);
-    $this->configuration['counter_orientation'] = $form_state->getValue(['settings', 'addthis_settings', 'counter_orientation']);
-    $this->configuration['extra_css'] = $form_state->getValue(['settings', 'addthis_settings', 'extra_css']);
+
+  public function addthisAjaxCallback(array $form, FormStateInterface $form_state){
+    return $form['settings']['settings']['addthis_settings']['type_settings'];
+  }
+
+  public function blockSubmit($form, FormStateInterface $form_state)
+  {
+    $this->configuration['type'] = $form_state->getValue(['settings', 'addthis_settings', 'type']);
+    $this->configuration['basic_toolbox']['share_services'] = $form_state->getValue(['settings', 'addthis_settings', 'type_settings', 'basic_toolbox', 'share_services']);
+    $this->configuration['basic_toolbox']['buttons_size'] = $form_state->getValue(['settings', 'addthis_settings', 'type_settings', 'basic_toolbox', 'buttons_size']);
+    $this->configuration['basic_toolbox']['counter_orientation'] = $form_state->getValue(['settings', 'addthis_settings', 'type_settings', 'basic_toolbox', 'counter_orientation']);
+    $this->configuration['basic_toolbox']['extra_css'] = $form_state->getValue(['settings', 'addthis_settings', 'type_settings', 'basic_toolbox', 'extra_css']);
+    $this->configuration['basic_button']['button_size'] = $form_state->getValue(['settings', 'addthis_settings', 'type_settings', 'basic_button', 'button_size']);
+    $this->configuration['basic_button']['extra_css'] = $form_state->getValue(['settings', 'addthis_settings', 'type_settings', 'basic_button', 'extra_css']);
+
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build()
+  {
```

Comment:
This code formatting is not going to pass. No big deal right now, but should note this moving forward